pandas/io/packers.py

```diff
@@ -253,23 +234,20 @@ def unconvert(values, dtype, compress=None):
     if dtype == np.object_:
         return np.array(values, dtype=object)
 
-    if compress == 'zlib':
+    values = values.encode('latin1')
 
+    if compress == 'zlib':
+        import zlib
         values = zlib.decompress(values)
         return np.frombuffer(values, dtype=dtype)
 
     elif compress == 'blosc':
-
-        if not _BLOSC:
-            raise Exception("cannot uncompress w/o blosc")
-
-        # decompress
+        import blosc
         values = blosc.decompress(values)
-
         return np.frombuffer(values, dtype=dtype)
 
     # from a string
-    return np.fromstring(values.encode('latin1'), dtype=dtype)
+    return np.fromstring(values, dtype=dtype)
```

Comment:
Actually, I moved the encode higher up to line 237.  For some reason, type(values) was unicode.  Both zlib.decompress and blosc.decompress do not take unicode and only accept encoded strings.